keycloak_authz_permission incorrectly sends a one element array as string

### Summary

When running:

```yaml
- name: Allow all users to list sessions
  community.general.keycloak_authz_permission:
    validate_certs: false
    name: allow-users-to-list-sessions
    state: present
    description: Allow users to list sessions
    permission_type: scope
    decision_strategy: AFFIRMATIVE
    scopes:
      - urn:ebo:CS:scopes:list
    resources:
      - Session
    policies:
      - all-users
    client_id: charging-stations
    realm: "{{ keycloak_realm }}"
    auth_keycloak_url: https://{{ ansible_default_ipv4.address }}:{{ keycloak_https_port }}/auth
    auth_username: admin
    auth_password: "{{ secret_keycloak_admin_password }}"
    auth_realm: master
```
It returns a 500, where java/keycloak prints:

```text
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.HashSet` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('34a556e1-95c0-482c-91df-4bccc865d356')
```

Which means it expects an array, but it got a string.

Looking at the python code [https://github.com/ansible-collections/community.general/blob/7721420388aa1d9cf7751fa250754d3419f3a2b1/plugins/modules/keycloak_authz_permission.py#L333](https://github.com/ansible-collections/community.general/blob/7721420388aa1d9cf7751fa250754d3419f3a2b1/plugins/modules/keycloak_authz_permission.py#L333):
```python
                payload['resources'] = r['_id']
```
I think this should be wrapped in an array.

    

### Issue Type

Bug Report

### Component Name

community.general.keycloak_authz_permission

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.15.3]
  config file = None
  configured module search path = ['/home/rgoussey/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/rgoussey/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/rgoussey/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/rgoussey/.local/bin/ansible
  python version = 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True

```


### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
Collection        Version
----------------- -------
community.general 7.3.0  

```


### Configuration

```console (paste below)

```


### OS / Environment

Ubuntu

### Steps to Reproduce

<!--- Paste example playbooks or commands between quotes below -->

Tweak variables, and run against a running keycloak.

```yaml (paste below)
- name: Allow all users to list sessions
  community.general.keycloak_authz_permission:
    validate_certs: false
    name: allow-users-to-list-sessions
    state: present
    description: Allow users to list sessions
    permission_type: scope
    decision_strategy: AFFIRMATIVE
    scopes:
      - urn:ebo:CS:scopes:list
    resources:
      - Session
    policies:
      - all-users
    client_id: charging-stations
    realm: "{{ keycloak_realm }}"
    auth_keycloak_url: https://{{ ansible_default_ipv4.address }}:{{ keycloak_https_port }}/auth
    auth_username: admin
    auth_password: "{{ secret_keycloak_admin_password }}"
    auth_realm: master
```


### Expected Results

The task should use the correct serialization format.

### Actual Results

```console (paste below)

```


### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

I can confirm this error exists with version 15.0.2.
```
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.util.HashSet` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('e0d9706f-fa20-4e25-8eab-f8e920e15436')
 at [Source: (String)"{"name": "view.members.permission.group.f90a06ec-3d04-4e2e-8e43-e4faec6dc4a8", "description": null, "type": "scope", "decisionStrategy": "AFFIRMATIVE", "logic": "POSITIVE", "scopes": ["28c9998c-a59a-443f-8af4-e6b1a29f9d26"], "resources": "e0d9706f-fa20-4e25-8eab-f8e920e15436", "policies": ["43fe2363-8ab9-4e39-a417-5765494110d7"], "id": "b04b1ad1-d30f-4a06-ad17-a8b8d7eb0e88"}"; line: 1, column: 239] (through reference chain: org.keycloak.representations.idm.authorization.ScopePermissionRepresentation["resources"])
```

--- a/changelogs/fragments/7151-fix-keycloak_authz_permission-incorrect-resource-payload.yml
+++ b/changelogs/fragments/7151-fix-keycloak_authz_permission-incorrect-resource-payload.yml
@@ -1,2 +1,2 @@
 bugfixes:
-  - keycloak_authz_permission - resource payload variable for scope-based permission was constructed as a string, when it needs to be a list, even for a single item. (https://github.com/ansible-collections/community.general/issues/7151).
+  - keycloak_authz_permission - resource payload variable for scope-based permission was constructed as a string, when it needs to be a list, even for a single item (https://github.com/ansible-collections/community.general/issues/7151).

--- a/changelogs/fragments/7151-fix-keycloak_authz_permission-incorrect-resource-payload.yml
+++ b/changelogs/fragments/7151-fix-keycloak_authz_permission-incorrect-resource-payload.yml
@@ -1,0 +1,2 @@
+bugfixes:
+  - keycloak_authz_permission - resource payload variable for scope-based permission was constructed as a string, when it needs to be a list, even for a single item. (https://github.com/ansible-collections/community.general/issues/7151).

--- a/plugins/modules/keycloak_authz_permission.py
+++ b/plugins/modules/keycloak_authz_permission.py
@@ -330,7 +330,7 @@ def main():
             if not r:
                 module.fail_json(msg='Unable to find authorization resource with name %s for client %s in realm %s' % (resources[0], cid, realm))
             else:
-                payload['resources'] = r['_id']
+                payload['resources'].append(r['_id'])
 
             for rs in r['scopes']:
                 resource_scopes.append(rs['id'])

